assets/course-theme/complete-lesson-button.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Initializes complete lesson transition.
+ */
+export const initCompleteLessonTransition = () => {
+	domReady( () => {
+		const completeForms = document.querySelectorAll(
+			'[data-id="complete-lesson-form"], .lesson_button_form'
+		);
+		const completeButtons = document.querySelectorAll(
+			'[data-id="complete-lesson-button"], .wp-block-sensei-lms-button-complete-lesson button'
+		);
+		const progressBars = document.querySelectorAll(
+			'.sensei-course-theme-course-progress-bar-inner'
+		);
+		const mainContent = document.querySelector(
+			'.sensei-course-theme__main-content'
+		);
+
+		/**
+		 * Disable complete buttons.
+		 */
+		const disableButtons = () => {
+			completeButtons.forEach( ( button ) => {
+				button.setAttribute( 'disabled', 'disabled' );
+				button.classList.add( 'is-busy' );
+			} );
+		};
+
+		/**
+		 * Delay submit to show animations.
+		 *
+		 * @param {Object} e    The submit event.
+		 * @param {Object} form The form DOM node.
+		 */
+		const delayFormSubmit = ( e, form ) => {
+			e.preventDefault();
+			disableButtons();
+
+			setTimeout( () => {
+				form.submit();
+			}, 1000 );
+		};
+
+		/**
+		 * Run progress bar animation.
+		 */
+		const runProgressBarAnimation = () => {
+			progressBars.forEach( ( progressBar ) => {
+				const { completed, count } = progressBar.dataset;
+
+				// Percentage with one more completed.
+				const percentage = ( ( +completed + 1 ) / +count ) * 100;
+				progressBar.style.width = `${ percentage }%`;
+			} );
+		};
+
+		/**
+		 * Form submit handler.
+		 *
+		 * @param {Object} e The submit event.
+		 */
+		const onFormSubmit = ( e ) => {
+			const form = e.target;
+
+			// Skip if the form is not for complete lesson (Reset lesson block, for example).
+			if (
+				! form.querySelector(
+					'input[name="quiz_action"][value="lesson-complete"]'
```

Comment:
Maybe not a better approach but: have you considered using this selector to identify the forms? I mean something like getting all these inputs and then looking for the parent form? So that we only hook to the correct form submit events.

I am quite muggle in terms of css/js so I am not sure it can be done or if it is worth the effort. Feel free to ignore the proposal :)

Reply:
Thank you for the idea Aarón!

But in this case it would be less performant, and with more code because currently we don't have a good selector for that. We'd need to get the buttons selector, then loop through them getting the form parent.

In the future, hopefully we'll be able to use this selector: https://caniuse.com/css-has, so we could select the forms that has the button inside. =)

Reply:
I thought a little more about this after see [another case](https://github.com/Automattic/sensei/pull/4546) that also needs to manipulate these elements, and I created this PR: https://github.com/Automattic/sensei/pull/4559 😉